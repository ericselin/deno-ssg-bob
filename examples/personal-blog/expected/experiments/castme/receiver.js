// namespace for messages
import { NAMESPACE } from "./shared.js";

// helper variables
const cast = window.cast;
const context = cast.framework.CastReceiverContext.getInstance();

// on custom messages, show the message in the `h1`
context.addCustomMessageListener(NAMESPACE, function(e) {
  // hide explainer paragraph
  document.querySelector("p").style.display = "none";
  // show the text from the data sent
  document.querySelector("h1").innerText = e.data.text;
});

// start the receiver context
context.start({
  // do not close if no activity (see docs)
  disableIdleTimeout: true
});
