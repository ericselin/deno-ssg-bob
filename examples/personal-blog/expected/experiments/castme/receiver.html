<!DOCTYPE html>
<html lang="en" class="receiver">
  <head>
    <title>Cast me</title>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link
      href="https://fonts.googleapis.com/css?family=Josefin+Sans&display=swap"
      rel="stylesheet"
    />
    <link rel="stylesheet" href="style.css" />
    <script
      type="text/javascript"
      src="//www.gstatic.com/cast/sdk/libs/caf_receiver/v3/cast_receiver_framework.js"
    ></script>
    <script src="receiver.js" type="module"></script>
  </head>
  <body>
    <p>
      Send messages from your desktop, and they'll appear here
    </p>
    <h1></h1>
  </body>
</html>
